aries_vcx_core/src/ledger/indy_vdr_ledger.rs

```diff
@@ -11,22 +11,22 @@ use vdr::ledger::requests::schema::{Schema, SchemaV1};
 use async_trait::async_trait;
 use serde_json::Value;
 use vdr::ledger::identifiers::{CredentialDefinitionId, RevocationRegistryId, SchemaId};
-use vdr::ledger::requests::{author_agreement::TxnAuthrAgrmtAcceptanceData, cred_def::CredentialDefinition};
+use vdr::ledger::requests::cred_def::CredentialDefinition;
 use vdr::ledger::RequestBuilder;
-use vdr::pool::{LedgerType, PreparedRequest, ProtocolVersion};
+use vdr::pool::{LedgerType, PreparedRequest, ProtocolVersion as VdrProtocolVersion};
 use vdr::utils::did::DidValue;
 use vdr::utils::Qualifiable;
 
 use crate::common::ledger::transactions::verify_transaction_can_be_endorsed;
-use crate::errors::error::VcxCoreResult;
-use crate::global::author_agreement::get_txn_author_agreement;
-use crate::global::settings;
+use crate::errors::error::{AriesVcxCoreError, AriesVcxCoreErrorKind, VcxCoreResult};
 
 use super::base_ledger::{AnoncredsLedgerRead, AnoncredsLedgerWrite, IndyLedgerRead, IndyLedgerWrite};
 use super::request_signer::RequestSigner;
 use super::request_submitter::RequestSubmitter;
 use super::response_cacher::ResponseCacher;
 
+// TODO: Should implement builders for these configs...
+// Good first issue?
```

Comment:
I am curious what @bobozaur @gmulhearn you guys think about `IndyVdrLedgerReadConfig`, `IndyVdrLedgerWriteConfig` in general.

Reply:
I am a little confused because I am not entirely sure what exactly is being asked by either of you. The context here is that there was a single small config struct here in the beginning. It was split into two and multiple fields were added. As their complexity grows, I just wanted to simplify their construction and so I wrote this comment. This was just the most simple and straightforward thing to do.

I have no issue with doing a builder pattern for the target struct, it just seemed like an overkill in the beginning.

I have a feeling that there is a miscommunication happening here and @Patrik-Stas incorrectly assumes my position is always and in all cases of a config struct aficionado, whereas I only said a few times that I prefer a single config object as an argument instead of a separate argument for each field resulting in functions taking zillion arguments :)

Reply:
Fair enough. I just think we should take a note of this and maybe create an issue when this is merged. Whether you wanna do it now or in a later refinement, I think they would be both fine. Replacing the config structs with builders would also be a good first issue, I think.

Reply:
it's not a biggie, just wanted to know other opinions, as we've discussed the similar things in past few times.

Also the term "a config struct aficionado" really got me lolled :-D